### PR TITLE
ARUHA-2114 Comparison operation for schema changes should always work on non-null values

### PR DESCRIPTION
**Pull request by antban**
ARUHA-2114 Comparison operation for schema changes should always work on non-null values

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/domain/SchemaChange.java [hunk 1]
```diff
@@ -2,26 +2,42 @@
...
+        PROPERTY_REMOVED(Version.Level.MAJOR),
+        DEPENDENCY_ARRAY_CHANGED(Version.Level.MAJOR),
+        DEPENDENCY_SCHEMA_CHANGED(Version.Level.MAJOR),
+        COMPOSITION_METHOD_CHANGED(Version.Level.MAJOR),
+        ATTRIBUTE_VALUE_CHANGED(Version.Level.MAJOR),
+        ENUM_ARRAY_CHANGED(Version.Level.MAJOR),
+        SUB_SCHEMA_CHANGED(Version.Level.MAJOR),
+        DEPENDENCY_SCHEMA_REMOVED(Version.Level.MAJOR),
+        REQUIRED_ARRAY_CHANGED(Version.Level.MAJOR),
+        REQUIRED_ARRAY_EXTENDED(Version.Level.MINOR, Version.Level.MAJOR),
```
Comment (rcillo) [inline src/main/java/org/zalando/nakadi/domain/SchemaChange.java]: `REQUIRED_ARRAY_EXTENDED` used to be `MINOR` for `forward` but `major` for `compatible`.

Comment (antban) [reply src/main/java/org/zalando/nakadi/domain/SchemaChange.java]: @rcillo Is it changed?

Comment (antban) [reply src/main/java/org/zalando/nakadi/domain/SchemaChange.java]: I mean it is exactly as it was: MINOR for forward and MAJOR for compatible.

Comment (rcillo) [reply src/main/java/org/zalando/nakadi/domain/SchemaChange.java]: 😞 my bad, I did not read it through.

**Comment by rcillo** [comment]
I did not actually understood where the problem was. Was there a change type missing for one of the compatibility modes?

**Comment by antban** [comment]
:+1:

**Pull request merged**